gateway/crates/federated-server/src/server/gateway.rs

```diff
@@ -40,6 +43,8 @@ pub(crate) struct GatewayConfig {
     pub rate_limit: Option<RateLimitConfig>,
     pub timeout: Option<Duration>,
     pub entity_caching: EntityCachingConfig,
+    pub config_hot_reload: bool,
+    pub config_path: Option<PathBuf>,
```

Comment:
:zap: **Suggestion** 
Ensure that the `config_path` is validated before being used to avoid potential runtime errors if the path is invalid.